src/providers/db2/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+
+########################################################
+# Files
+
+SET(DB2_SRCS
+  qgsdb2provider.cpp qgsdb2dataitems.cpp qgsdb2newconnection.cpp qgsdb2tablemodel.cpp qgsdb2featureiterator.cpp qgsdb2sourceselect.cpp
+)
+SET(DB2_MOC_HDRS
+  qgsdb2provider.h qgsdb2dataitems.h qgsdb2newconnection.h qgsdb2tablemodel.h qgsdb2featureiterator.h qgsdb2sourceselect.h
```

Comment:
qgsdb2provider.h and qgsdb2featureiterator.h shouldn't be listed here - only headers with the Q_OBJECT macro need to be included in this section. That's what's causing the build errors.